fastly/resource_fastly_service_dictionary_items_v1.go

```diff
@@ -0,0 +1,219 @@
+package fastly
+
+import (
+	"fmt"
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceServiceDictionaryItemsV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServiceDictionaryItemsV1Create,
+		Read:   resourceServiceDictionaryItemsV1Read,
+		Update: resourceServiceDictionaryItemsV1Update,
+		Delete: resourceServiceDictionaryItemsV1Delete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"service_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Service Id",
+			},
+
+			"dictionary_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Dictionary Id",
+			},
+
+			"items": {
+				Type:        schema.TypeMap,
+				Description: "Dictionary Items",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceServiceDictionaryItemsV1Create(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*FastlyClient).conn
+
+	serviceID := d.Get("service_id").(string)
+	dictionaryID := d.Get("dictionary_id").(string)
+	items := d.Get("items").(map[string]interface{})
+
+	for k, v := range items {
+		_, err := conn.CreateDictionaryItem(&gofastly.CreateDictionaryItemInput{
```

Comment:
Instead of looping over the items and calling `CreateDictionaryItem` for each item, it would be more efficient to use the [batch API endpoint](https://docs.fastly.com/api/config#dictionary_item_dc826ce1255a7c42bc48eb204eed8f7f) and use [`CreateDictionaryItems`](https://godoc.org/github.com/fastly/go-fastly/fastly#Client.CreateDictionaryItems) instead.

Reply:
The Create, Update and Delete functions now use batch operations to alter the fastly remote state.  The Read function still uses the list function.